File: 정종길/프로그래머스/로또의 최고 순위와 최저 순위.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer;
        int right = 0;
        List<Integer> li = Arrays.stream(win_nums).boxed().collect(Collectors.toList());

        int cnt = count(0, lottos);

        for(int num : lottos){
            if(li.indexOf(num) != -1){
                right += 1;
            }
        }

        if(right == 0){
            if(cnt == 0){
                answer = new int[] {6, 6};
            }
            else{
                answer = new int[] {7 - (cnt), 6};
            }

        }else {
            answer = new int[] {7 - (right + cnt), 7 - right};
        }
        return answer;
    }

    public int count(int num, int[] lottos){
        int cnt = 0;
        for(int lotto : lottos){
            if(num == lotto){
                cnt += 1;
            }
        }
        return cnt;
    }
}

File: 정종길/프로그래머스/카카오 2회독/2021 Dev-Matching 웹 백엔드 개발자(상반기)/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    win_nums = set(win_nums)
    zero = win = 0
    rank = { i : (7 - i if i != 0 else 6) for i in range(7) }
    for num in lottos:
        if num == 0:
            zero += 1
            continue
            
        if num in win_nums:
            win += 1
            win_nums.remove(num)
    
    answer = [rank[win + min(zero, len(win_nums))], rank[win]]
    
    return answer